addons/web/static/src/js/model.js

```diff
@@ -417,10 +424,42 @@ odoo.define("web/static/src/js/model.js", function (require) {
          * @returns {Promise}
          */
         async load() {
-            for (const extension of this.extensionsToLoad) {
-                await extension.load();
+            const toLoad = new Map();
+            const loaded = [];
+            // Determine which extensions need to be loaded
+            for (const extension of this.extensions) {
+                const { shouldLoad, dependencies } = extension.getLoadingState();
+                if (shouldLoad) {
+                    for (const dep of dependencies) {
+                        if (!this.extensions.some((e) => e.name === dep)) {
+                            // Any dependency not found in the registered
```

Comment:
why? I think it is not a good idea conceptually

Reply:
Done